modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/RotateGestureRecognizer.java

```diff
@@ -231,7 +227,7 @@ public void notifyEndTouchEvent(long time) {
                 sendRotateFinishedEvent();
             }
             if (ROTATION_INERTIA_ENABLED && (state == RotateRecognitionState.PRE_INERTIA || state == RotateRecognitionState.ACTIVE)) {
-                double timeFromLastRotation = ((double)time - rotationStartTime) / 1000000;
+                double timeFromLastRotation = (time - rotationStartTime) / 1000000;
```

Comment:
Another place where I'm not sure removing the explicit cast is best (probably OK here).

Reply:
rotationStartTime is a double.  it is really unnecessary

Reply:
You are correct that with these kind of changes you can't see if it is correct just from looking at the diff. Variables would need to be named more explicitly, or explicit casts would need to be added to repeat the type information. If the goal is to make it clear in the diff, then many places would benefit from explicit casts (probably in more places than where they were removed).  I think the general trend is to avoid explicit casts, but I could be wrong.

I can revert these changes and we can disable the warning.

Specifically about this code, `rotationStartTime` probably should not have been a `double` depending on what is stored in it (checking: its source is `System.nanoTime()`) as the conversion from `long` to `double` will lose precision in the area where it really counts as it wants to see the difference of the current and last value, so in this case the warning may have exposed a (precision) bug.

Reply:
+1 for reverting the changes and disabling the warning

Reply:
I've reverted the changes to the gesture recognizers and filed https://bugs.openjdk.org/browse/JDK-8298060